addons/social_share/wizards/snshare_url_share.py

```diff
@@ -0,0 +1,60 @@
+from odoo import api, fields, models
+from ..controllers.main import SocialShareController
+from ..utils.image_utils import scale_image_b64
+
+
+class SocialShareUrlShare(models.TransientModel):
+    _name = 'snshare.url.share'
+    _description = 'Social Share URL Wizard'
+
+    campaign_id = fields.Many2one('snshare.campaign')
+    model_id = fields.Many2one('ir.model', related='campaign_id.model_id')
+    res_id = fields.Reference(selection="_selection_res_id", string="Record")
```

Comment:
Would probably differentiate the record, from res_id (res_ids / domain, if we go multi), to ease understanding. Generally we expect res_id to be an integer (or a reference integer).